src/time.cr

```diff
@@ -275,76 +281,90 @@ struct Time
     end
   end
 
-  def self.now : self
+  # Returns the current time in the current time zone.
+  def self.now : Time
     new
   end
 
-  def self.utc_now : self
+  # Returns the current time in UTC time zone.
+  def self.utc_now : Time
     seconds, nanoseconds = compute_seconds_and_nanoseconds
     new(seconds: seconds, nanoseconds: nanoseconds, kind: Kind::Utc)
   end
 
-  def date
+  # Returns a copy of `self` with time-of-day components (hour, minute, ...) set to zero.
+  def date : Time
     Time.new(year, month, day, kind: kind)
   end
 
-  def year
+  # Returns the year number (Common Era).
+  def year : Int32
     year_month_day_day_year[0]
   end
 
-  def month
+  # Returns the month number of the year (`1..12`).
+  def month : Int32
     year_month_day_day_year[1]
   end
 
-  def day
+  # Returns the day number of the month (`0..31`).
+  def day : Int32
     year_month_day_day_year[2]
   end
 
-  def hour
+  # Returns the hour of the day (`0...24`).
```

Comment:
The day is defined as 24 hours and it's also clearer that time goes up to almost 24, not just 23. I will not be making this change.
I also don't know about leap seconds, and whether they are even possible in this implementation.

Reply:
A leap second doesn't lead to `24:00:00`. It is represented as `23:59:60`. So the only field that *might*  have a irregular maximum value because of leap seconds is `second`.

And I'd strongly suggest to use `0..23` because this accurately represents the value range of `hour`. There is no "almost 24", it is not represented in this granularity. Times from `23:00:00` to `23:59:59`  (or `:60`) are all `hour 23` without any further distinction. There is no sense in using an exclusive range here.
Besides, the exact meaning of the exclusive range literal `...` is not as commonly known and easily understood as the inclusive range `..`

Reply:
I'm even more convinced this is bad now.

Reply:
ok